easybuild/toolchains/compiler/gcc.py

```diff
@@ -110,3 +117,43 @@ def _set_compiler_vars(self):
                 raise EasyBuildError("Failed to determine software root for GCC")
 
         self.variables.append_subdirs("LDFLAGS", gcc_root, subdirs=["lib64", "lib"])
+
+    def _set_optimal_architecture(self, default_optarch=None):
+        """
+        GCC-specific adjustments for optimal architecture flags.
+
+        :param default_optarch: default value to use for optarch, rather than using default value based on architecture
+                                (--optarch and --optarch=GENERIC still override this value)
+        """
+        if default_optarch is None and self.arch == systemtools.AARCH64:
+            # On AArch64, -mcpu=native is not supported prior to GCC 6.  In this case, try to optimize for the detected
+            # CPU model (vanilla ARM cores only).  Note that this heuristic may fail if the CPU model is not supported
+            # by the GCC version being used.
+            gcc_version = self.get_software_version(self.COMPILER_MODULE_NAME)[0]
```

Comment:
No, I propose to try obtaining the version using `GCCcore` first, and fall back to use `GCC` if that doesn't work. Whether you use `self.get_software_version` or `get_software_version` doesn't really matter here, that's only relevant when multiple modules are in play (`self.get_software_version` has a way of dealing with that).

So, something like this:

```python
gcc_ver = get_software_version('GCCcore')
if gcc_ver is None:
    gcc_ver = get_software_version('GCC')
```

(switching to `get_software_version` makes sense here since we know we'll never be dealing with multiple names at the same time here)